src/main/java/Duke/Parser.java

```diff
@@ -0,0 +1,133 @@
+                return new DelCommand(taskID);
+            } catch (NumberFormatException e) {
+                throw new InvalidCommandException(INVALID_NUMBER_FORMAT_MSG);
+            }
+        } else if (input.startsWith("todo")) {
+            errMsg = "☹ OOPS!!! The description of a todo task cannot be empty!";
+            String taskDescription = splitWithFormat(input, " ", errMsg)[1];
+            return new AddCommand(new ToDoTask(taskDescription));
+        } else if (input.startsWith("event")) {
+            errMsg = "☹ OOPS!!! I need more details on the event!\n"
+                    + "(Format: event _description_ /from _start_time_ /to _end_time_)";
+            String taskDetailsStr = splitWithFormat(input, " ", errMsg)[1];
+            String[] taskDetails = splitWithFormat(taskDetailsStr, "/from", errMsg);
+            String taskDesription = taskDetails[0];
+            String[] taskPeriod = splitWithFormat(taskDetails[1], "/to", errMsg); 
+            try {
+                LocalDateTime taskStart = LocalDateTime.parse(taskPeriod[0], DATETIME_FORMATTER);
+                LocalDateTime taskEnd =  LocalDateTime.parse(taskPeriod[1], DATETIME_FORMATTER);
+                return new AddCommand(new EventTask(taskDesription, taskStart, taskEnd));
+            } catch (DateTimeParseException e) {
+                throw new InvalidCommandException(INVALID_TIME_FORMAT_MSG);
+            }
+        } else if (input.startsWith("deadline")) {
+            errMsg = "☹ OOPS!!! I need more details on the event!\n"
+            + "(Format: deadline _description_ /by _time_)";
+            String taskDetailsStr = splitWithFormat(input, " ", errMsg)[1];
+            String[] taskDetails = splitWithFormat(taskDetailsStr, "/by", errMsg);
+            String taskDesription = taskDetails[0]; 
+            try {
+                LocalDateTime deadline = LocalDateTime.parse(taskDetails[1], DATETIME_FORMATTER);
+                return new AddCommand(new DeadlineTask(taskDesription, deadline));
+            } catch (DateTimeParseException e) {
+                throw new InvalidCommandException(INVALID_TIME_FORMAT_MSG);
+            }
+        } else if (input.startsWith("find")) {
+            errMsg = "";
+            String keyword = splitWithFormat(input, " ", errMsg)[1];
+            return new FindCommand(keyword);
+        } else {
+            throw new InvalidCommandException("☹ OOPS!!! I dont recognise this command!");
+        }
+    }
+
+    /**
+     * Checks that the string can be splitted and split it to fit the command.
+     * 
+     * @param input The user input line to be delimited.
+     * @param delimiter The delimiter used to delimit the command.
+     * @param errMsg The error message associated with the given command.
+     * @return The splitted string.
+     * @throws InvalidCommandException
+     */
+    private static String[] splitWithFormat(String input, String delimiter, String errMsg) 
```

Comment:
Good abstraction here haha 👍 

Can possibly abstract away the max number of splits too (e.g. in this case, 2)